elyra/pipeline/kfp/processor_kfp.py

```diff
@@ -1058,11 +1059,21 @@ def _compose_container_command_args(
         account whether the container will run in a CRI-O environment.
         """
         elyra_github_org = os.getenv("ELYRA_GITHUB_ORG", "elyra-ai")
-        elyra_github_branch = os.getenv("ELYRA_GITHUB_BRANCH", "main" if "dev" in __version__ else "v" + __version__)
+        elyra_github_branch = os.getenv("ELYRA_GITHUB_BRANCH", "dspv2" if "dev" in __version__ else __version__)
```

Comment:
> if "dev" in __version__ else __version__
1. since there is no "v" anymore, does it make sense to have this `if` statement here or I simply don't understand the code?
1. since we removed the "v" anymore, this isn't backward compatible with the upstream elyra-ai repository -> thus why do we bother with it on lines 1063-1066?
** if we care about backward compatibility, then we should count with the "v" prefix in a proper situation, I think

Reply:
The motivation is to use the the bootstrapper from the default branch if the version points to a dev release.

Reply:
okay, good - understand point 1 now
but then the point 2 is still an issue, though, see:

* https://github.com/elyra-ai/elyra/releases
* vs
* https://github.com/opendatahub-io/elyra/releases

to be backward compatible, we should have this on line 1065, I think:

```
https://raw.githubusercontent.com/{elyra_github_org}/elyra/v{elyra_github_branch}/elyra/kfp/bootstrapper.py
```

the `v` added to the branch/release - haven't check what other implications this may have though

Reply:
This part doesnt need to be backward compatible directly for two reasons:
1. As this odh-elyra part, the user can point their choice of URL , by setting env var `ELYRA_BOOTSTRAP_SCRIPT_URL` , we are only changing the default to point to this fork.
2. As this v2 work, lot of the work needs to point to updated v2 runtime bootstrapper URL.

Reply:
Okay, I got it now, Harshad, thanks - I suppose I somehow forgot we're in the elyra repo here and was thinking in the context of kubeflow repo where we need to still take into account the v1 pipelines. My fault. Thanks again :slightly_smiling_face:

Reply:
All good, thanks :)